src/compiler/scala/tools/nsc/backend/jvm/BTypesFromSymbols.scala

```diff
@@ -592,11 +578,10 @@ class BTypesFromSymbols[G <: Global](val global: G) extends BTypes {
 
   /**
    * True for module classes of modules that are top-level or owned only by objects. Module classes
-   * for such objects will get a MODULE$ flag and a corresponding static initializer.
+   * for such objects will get a MODULE$ field and a corresponding static initializer.
    */
   final def isStaticModuleClass(sym: Symbol): Boolean = {
-    /* (1) Phase travel to to pickler is required to exclude implementation classes; they have the
-     * lateMODULEs after mixin, so isModuleClass would be true.
+    /* (1) Phase travel to pickler to exclude the lateMODULEs flag.
```

Comment:
yeah i was wondering and didn't dare.. but looking again at the details again it should be fine, i'll give it a try.